BDArmory/Weapons/BDExplosivePart.cs

```diff
@@ -164,6 +166,7 @@ public override void OnStart(StartState state)
             */
             CalculateBlast();
             ParseWarheadType();
+            direction = part.partTransform.forward;
```

Comment:
Is there a reason for setting this here? It's setting the direction to that of the part when it was first spawned, which doesn't really make sense.
I see the direction gets set within `Detonate`, but not `DetonateIfPossible`, presumably this is so that `MissileLauncher` can set the direction on line 2181 of MissileLauncher.cs, however this means that the direction is wrong if it gets triggered via `OnJustAboutToBeDestroyed` or `BDModularGuidance.Detonate`. Also, for standard warheads, the direction from line 2181 of MissileLauncher.cs is getting overwritten.
It might be better to have something like
```csharp
public void DetonateIfPossible()
{
    DetonateIfPossible(warheadType == "standard" ? default: part.partTransform.forward);
}
public void DetonateIfPossible(Vector3 towards)
{
    ...
    direction = towards;
    ...
}
```
and have `MissileLauncher.Detonate` (and maybe `BDModularGuidance.Detonate`) call the latter while `OnJustAboutToBeDestroyed` calls the former.

Reply:
No. you are correct, that should be elsewhere.
```
public void Detonate( IfPossible() )
{
    . . .
    direction = part.partTransform.forward (or we could do direction = (warheadType == "standard" ? default: part.partTransform.forward);)
    . . .
}
```
as both the small warhead model's transform and the missileReferenceTransform forward direction is Z+.
Setting the direction to Default if a standard warhead was due to Explosion only needing a direction vector for the shaped/annular blast patterns, and using direction = default as a shortcut for standard warheads in `IsAngleAllowed()` to skip a Vector3.Angle check.